### ISSUE TITLE
v0.13.2 error: packaging.specifiers.InvalidSpecifier

### ISSUE THREAD

**Issue by shamilbi** [ido50/morgan#11]
morgan-0.13.2, python-3.11, linux-x86_64

morgan.ini:
[requirements]
flit =

morgan mirror

output (only in v0.13.2):
```
flit
Traceback (most recent call last):
  File "~/apps/morgan-0.13.2/.venv/lib64/python3.11/site-packages/packaging/specifiers.py", line 634, in __init__
    parsed.add(Specifier(specifier))
               ^^^^^^^^^^^^^^^^^^^^
  File "~/apps/morgan-0.13.2/.venv/lib64/python3.11/site-packages/packaging/specifiers.py", line 98, in __init__
    raise InvalidSpecifier(f"Invalid specifier: '{spec}'")
packaging.specifiers.InvalidSpecifier: Invalid specifier: '3'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "~/apps/morgan/.venv/bin/morgan", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "~/apps/morgan-0.13.2/.venv/lib64/python3.11/site-packages/morgan/__init__.py", line 472, in main
    mirror(args.index_path)
  File "~/apps/morgan-0.13.2/.venv/lib64/python3.11/site-packages/morgan/__init__.py", line 435, in mirror
    m.mirror("{}{}".format(package, m.config["requirements"][package]))
  File "~/apps/morgan-0.13.2/.venv/lib64/python3.11/site-packages/morgan/__init__.py", line 74, in mirror
    deps = self._mirror(requirement)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/apps/morgan-0.13.2/.venv/lib64/python3.11/site-packages/morgan/__init__.py", line 150, in _mirror
    files = self._filter_files(requirement, files)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/apps/morgan-0.13.2/.venv/lib64/python3.11/site-packages/morgan/__init__.py", line 232, in _filter_files
    files = list(filter(
            ^^^^^^^^^^^^
  File "~/apps/morgan-0.13.2/.venv/lib64/python3.11/site-packages/morgan/__init__.py", line 233, in <lambda>
    lambda file: self._matches_environments(file), files))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/apps/morgan-0.13.2/.venv/lib64/python3.11/site-packages/morgan/__init__.py", line 250, in _matches_environments
    spec_set = packaging.specifiers.SpecifierSet(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/apps/morgan-0.13.2/.venv/lib64/python3.11/site-packages/packaging/specifiers.py", line 636, in __init__
    parsed.add(LegacySpecifier(specifier))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/apps/morgan-0.13.2/.venv/lib64/python3.11/site-packages/packaging/specifiers.py", line 253, in __init__
    super().__init__(spec, prereleases)
  File "~/apps/morgan-0.13.2/.venv/lib64/python3.11/site-packages/packaging/specifiers.py", line 98, in __init__
    raise InvalidSpecifier(f"Invalid specifier: '{spec}'")
packaging.specifiers.InvalidSpecifier: Invalid specifier: '3'
```

**Comment by ido50** [comment]
> flit =

Is the requirement empty on purpose?

**Comment by shamilbi** [comment]
> Is the requirement empty on purpose?
Yes, last version

**Comment by ido50** [comment]
Ha, I don't think I've even intended for this to be possible, but see no reason why I shouldn't. I'll see if I can fix this.

**Comment by ido50** [comment]
Yeah, I think it worked beforehand by accident :) Still, I'll fix this.

**Comment by ido50** [comment]
Can you provide your environment configuration? I can't seem to replicate.

**Comment by shamilbi** [comment]
[env.None]
os_name = posix
sys_platform = linux
platform_machine = x86_64
platform_python_implementation = CPython
platform_system = Linux
python_version = 3.11
python_full_version = 3.11.5
implementation_name = cpython

**Comment by shamilbi** [comment]
it worked on 'flit = >=3'

**Comment by shamilbi** [comment]
error on:
flit = <1

**Comment by ido50** [comment]
Just a quick update that I found and fixed the issue, but I've been having a lot of trouble testing for a few days now because the PyPI API is taking _a lot_ of time to respond to requests, if at all. I'm not sure if this problem is only on my end or not, but it's really unresponsive. I'll keep trying, I don't want to release without testing, even though it's an easy fix.

**Comment by ido50** [comment]
I've been able to get responses when connecting to different VPNs, that's weird. I should be able to release a fix soon.

**Comment by ido50** [comment]
Fixed in v0.13.3. By the way, I miss-remembered my own program, I did intend for Morgan to download the latest available version if an empty specifier was given, I forgot about that feature because I never use it :)
